### ISSUE TITLE
Allow multibinding with @Binds provision methods

### ISSUE THREAD

**Issue by tadfisher** [google/dagger#367]
If I attempt to combine `@Binds` and, for example, `@IntoSet`:

```java
@Binds @IntoSet @PerSession
abstract RequestHandler providePicassoRequestHandler(UserFaceRequestHandler requestHandler);
```

Dagger fails with:

```
error: Multibinding annotations may only be on @Provides or @Produces methods
```

Is there a design reason for this restriction, or did the compiler check not get updated?

**Comment by gk5885** [comment]
This is absolutely in the works.  It's requiring a little bit of a reimplementation of set multibindings (they're all getting more efficient!), but @ronshapiro is on it.

**Comment by ronshapiro** [comment]
This was released in `2.5`
